### PR TITLE
Sema: Continue to hollow out SolverScope

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/Sema/ConstraintSystem.cpp [hunk 1]
```diff
@@ -1278,10 +1276,10 @@ Type ConstraintSystem::openPackExpansionType(PackExpansionType *expansion,
 void ConstraintSystem::recordOpenedPackExpansionType(PackExpansionType *expansion,
                                                      TypeVariableType *expansionVar) {
   bool inserted = OpenedPackExpansionTypes.insert({expansion, expansionVar}).second;
-  if (inserted) {
-    if (isRecordingChanges())
-      recordChange(SolverTrail::Change::recordedOpenedPackExpansionType(expansion));
-  }
+  ASSERT(inserted);
+
+  if (solverState)
```
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:1281]: Could you remind me in what cases we don’t have a state available, I thought that it usually gets constructed very early, maybe some typeCheck* methods don’t have it initialized which might be a bug?…

Comment (slavapestov) [reply lib/Sema/ConstraintSystem.cpp:1281]: I've seen a few code paths where solverState was null, from CSGen I think.

**Pull request merged**